classes/class3/homework/submit_casts/app.js

```diff
@@ -0,0 +1,36 @@
+var express = require('express');
+var app = express();
+
+var path = require('path')
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var index = require('./routes/index.js')
+var exphbs = require('express-handlebars');
+var PORT = process.env.PORT || 3000;
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+
+app.get('/',function(req,res){
+	res.render("home",{'paths':['/casts','/casts/new','/casts/bymember/Adam Archer','/casts/delete/old']});
+});
+
+app.get('/casts', index.casts);
+app.get('/casts/new', index.newcast);
+app.get('/casts/bymember/*', index.bymember);
```

Comment:
This should be `'/casts/bymember/:member'`. The `*` is just going to accept any random string that follows that third `/`, including the URL encoded space character you have above. This is dangerous because it's allowing any arbitrary string sent by the user to query your database. It would also accept a route like `/casts/bymember/member_name/another/route/thing/here.html`, which can potentially break any routes you might have that look like `/casts/bymember/:member/name` or etc.

Just a thing to keep in mind going forward in this class.